lib/Coffer/Path.hs

```diff
@@ -177,14 +196,30 @@ data QualifiedPath path = QualifiedPath
   { qpBackendName :: Maybe BackendName
   , qpPath :: path
   }
-  deriving stock (Show, Eq, Functor)
+  deriving stock (Show, Eq, Functor, Generic)
+  deriving anyclass (FromJSON, ToJSON)
 
 instance (Buildable path) => Buildable (QualifiedPath path) where
   build (QualifiedPath backendNameMb path) =
     case backendNameMb of
       Just backendName -> build backendName <> "#" <> build path
       Nothing -> build path
 
+instance (FromHttpApiData path) => FromHttpApiData (QualifiedPath path) where
+  parseUrlPiece text
+    | [pathPiece] <- T.splitOn "#" text = do
+        path <- parseUrlPiece @path pathPiece
+        pure $ QualifiedPath Nothing path
+    | [backendPiece, pathPiece] <- T.splitOn "#" text = do
+        backendName <- newBackendName backendPiece
+        path <- parseUrlPiece @path pathPiece
+        pure $ QualifiedPath (Just backendName) path
+    | otherwise =
+        Left "Invalid qualified path format. Expected [BACKENDNAME#]PATH"
```

Comment:
This could be refactored to avoid using `T.splitOn "#"` twice. But don't worry about it, I've added a note in #89 to refactor it later.